core/src/main/java/hudson/Util.java

```diff
@@ -1927,4 +1927,14 @@ public static long daysElapsedSince(@NonNull Date date) {
     private static PathRemover newPathRemover(@NonNull PathRemover.PathChecker pathChecker) {
         return PathRemover.newFilteredRobustRemover(pathChecker, DELETION_RETRIES, GC_AFTER_FAILED_DELETE, WAIT_BETWEEN_DELETION_RETRIES);
     }
+
+    /**
+     * Returns Hex string of SHA-256 Digest of input bytes
+     */
+    public static String getHexStringOfSHA256DigestOf(@NonNull byte[] input) throws NoSuchAlgorithmException
```

Comment:
as this is not needed to be used externally it can be benificial to restrict it from plugins so we do not introduce a new API that can not be updated in the future.

```suggestion
    @Restricted(NoExternalUse.class)
    public static String getHexStringOfSHA256DigestOf(@NonNull byte[] input) throws NoSuchAlgorithmException
```

Also consider if this should be just creating the digest, and then let callers call `toHexString` if they want something printable.